### PR TITLE
youtube-dl plugin for totem

### PR DESCRIPTION
**Pull request by berniepark**

### CODE REVIEW

**Comment by hadess** [comment]
Looks like it would solve https://github.com/flathub/org.gnome.Totem/issues/38

**Comment by barthalion** [comment]
bot, build org.gnome.Totem.Videosite.YouTubeDl

File: org.gnome.Totem.Videosite.YouTubeDl.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+id: org.gnome.Totem.Videosite.YouTubeDl
+runtime: org.gnome.Totem
```
Comment (barthalion) [inline org.gnome.Totem.Videosite.YouTubeDl.yaml:2]:
```suggestion
runtime: org.gnome.Totem//stable
```

**Comment by barthalion** [comment]
And then, of course, we need a different download URL:

```
Downloading http://yt-dl.org/downloads/latest/youtube-dl-2020.11.01.1.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     3    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     3    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
Failed to download sources: module youtube-dl: The requested URL returned error: 451 Unavailable for Legal Reasons
```

**Comment by barthalion** [comment]
bot, build org.gnome.Totem.Videosite.YouTubeDl

File: org.gnome.Totem.Videosite.YouTubeDl.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    PIP_TARGET: /app/lib/totem-pl-parser/YouTubeDl/site-packages
+
+modules:
+  - name: youtube-dl
+    buildsystem: simple
+    build-commands:
+      - pip3 install .
+    sources:
+      - type: archive
+        url: http://yt-dl.org/downloads/latest/youtube-dl-2020.11.12.tar.gz
```
Comment (Erick555) [inline org.gnome.Totem.Videosite.YouTubeDl.yaml:19]: `https`?

File: org.gnome.Totem.Videosite.YouTubeDl.yaml (hunk 1 above)
Comment (berniepark) [reply org.gnome.Totem.Videosite.YouTubeDl.yaml]: that doesnt work, but builds on my system. is it a flathub problem?

Comment (hfiguiere) [reply org.gnome.Totem.Videosite.YouTubeDl.yaml]: You have to use "runtime-version" to specify the base app version for it to work on flathub.

Comment (hadess) [reply org.gnome.Totem.Videosite.YouTubeDl.yaml]: When I added extensions support to totem, I tried to make sure not to need that. Does that mean that flathub extensions would need to be respun for every release bump? That does sound like a problem as Bernie mentioned :/

Comment (barthalion) [reply org.gnome.Totem.Videosite.YouTubeDl.yaml]: I think `runtime-version` is equivalent of `//stable` notation but maybe I was wrong.

@hadess It's about flatpak branch, not runtime version. Unless it needs a rebuild every Totem release, that is.

Comment (hfiguiere) [reply org.gnome.Totem.Videosite.YouTubeDl.yaml]: I got the cue from: https://flathub.org/builds/#/builders/32/builds/32332/steps/7/logs/err_text
It is equivalent in `flatpak-builder` (hence working locally)

I ran into that when doing the audio plugins (I have a program to generate manifests)

Comment (hadess) [reply org.gnome.Totem.Videosite.YouTubeDl.yaml]: Ha, so: `runtime-version: stable` should work, correct? Good, nothing that I'd need to change on my side ;)

**Comment by barthalion** [comment]
bot, build org.gnome.Totem.Videosite.YouTubeDl

**Comment by barthalion** [comment]
It misses metainfo file. An example: https://github.com/flathub/org.gajim.Gajim.Plugin.appindicator/blob/master/org.gajim.Gajim.Plugin.appindicator.metainfo.xml (and https://github.com/flathub/org.gajim.Gajim.Plugin.appindicator/blob/master/flathub.json#L2)

**Comment by nedrichards** [comment]
bot, build org.gnome.Totem.Videosite.YouTubeDl

**Comment by bilelmoussaoui** [comment]
bot, build org.gnome.Totem.Videosite.YouTubeDl

File: org.gnome.Totem.Videosite.YouTubeDl.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        sha256: aeb00b2079c4711de7eea2a91f6923ebade84c5e985b0caeb47d572395d42831
+    cleanup:
+      - 'etc/'
+      - 'share/'
+
+  - name: totem-pl-parser-youtube-dl
+    buildsystem: simple
+    build-commands:
+      - install -Dm755 totem-ydl-parser.py ${FLATPAK_DEST}/bin/01-totem-pl-parser-videosite-youtube-dl
+      - install -m0644 -D org.gnome.Totem.Videosite.YouTubeDl.metainfo.xml ${FLATPAK_DEST}/share/metainfo/org.gnome.Totem.Videosite.YouTubeDl.metainfo.xml
```
Comment (bilelmoussaoui) [inline org.gnome.Totem.Videosite.YouTubeDl.yaml:30]: you need to call appstream-compose afterwards; see how other extensions do that

**Comment by bilelmoussaoui** [comment]
bot, build org.gnome.Totem.Videosite.YouTubeDl

**Pull request closed without merging**